pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -58,18 +58,25 @@ public ZKOperator(PinotHelixResourceManager pinotHelixResourceManager, Controlle
     _controllerMetrics = controllerMetrics;
   }
 
-  public void completeSegmentOperations(String tableNameWithType, SegmentMetadata segmentMetadata,
+  public boolean completeSegmentOperations(String tableNameWithType, SegmentMetadata segmentMetadata,
```

Comment:
We are using exception to pass the failure information for `processNewSegment()` and `processExistingSegment()`, so suggest directly throwing exception if refreshing segment does not exist. Directly throwing exception has the benefit of having more context info, and the accurate stack trace